### PR TITLE
feat: rename `dd:ignore` to `orchestrion:ignore`

### PR DESCRIPTION
**Pull request by RomainMuller**
The `//orchestrion:ignore` directive replaces `//dd:ignore` to make it clear that it opts the node tree out of any orchestrion processing, and not just of datadog integrations.

The `dd:ignore` directive continues to work, but a warning will be emitted when it is encountered, prompting users to update their code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,12 +127,12 @@ Library                                               | Since    | Notes
 `github.com/go-redis/redis/v8`                        | `v0.7.0` | [Aspect][go-redis-v8]
 `github.com/gofiber/fiber/v2`                         | `v0.7.0` | [Aspect][fiber-v2]
 `github.com/gomodule/redigo/redis`                    | `v0.7.0` | [Aspect][redigo]
-`github.com/gorilla/mux`                              | `v0.7.0` | [Aspect][gorilla]. Cannot be opted out of via `//dd:ignore`
+`github.com/gorilla/mux`                              | `v0.7.0` | [Aspect][gorilla] (library-side)
```
Comment (eliottness) [inline README.md:130]: How about adding a footnote to explain what library-side means

Comment (eliottness) [reply README.md:130]: Or linking to the doc of orchestrion:ignore

File: docs/content/docs/custom-trace.md [hunk 2]
```diff
@@ -19,8 +19,9 @@ to not be instrumented, either because they result in excessively verbose
 traces, or because those trace spans would be duplicated (for example, when
 using a custom-made `net/http` middleware stack).
 
-The `//dd:ignore` directive can be added anywhere in your application's code,
-and will disable all `orchestrion` instrumentation in the annotated syntax tree.
+The `//orchestrion:ignore` directive can be added anywhere in your application's
+code, and will disable all `orchestrion` instrumentation in the annotated syntax
+tree.
```
Comment (eliottness) [inline docs/content/docs/custom-trace.md:24]: Would you mind adding a warning saying that it won't work on some aspects of?

File: internal/injector/labels.go [hunk 3]
```diff
@@ -7,18 +7,29 @@ package injector
...
 )
 
-// ddIgnored returns true if the node is prefixed by a `//dd:ignore` directive.
-func ddIgnored(node dst.Node) bool {
+var warnOnce sync.Once
+
+// isIgnored returns true if the node is prefixed by an `//orchestrion:ignore` (or the legacy `//dd:ignore`) directive.
+func isIgnored(node dst.Node) bool {
 	for _, cmt := range node.Decorations().Start.All() {
+		if cmt == orchestrionIgnore || strings.HasPrefix(cmt, orchestrionIgnore+" ") {
```
Comment (eliottness) [inline internal/injector/labels.go:26]: Why not directly call strings.HasPrefix without the spacr at the end? Am I missing something?

**Review by eliottness — CHANGES_REQUESTED** [review]

File: internal/injector/labels.go (hunk 3 above)
Comment (RomainMuller) [reply internal/injector/labels.go:26]: Because `//orchestrion:ignoreItAllPlease` is not `//orchestrion:ignore`.

**Review by eliottness — APPROVED** [review]

**Pull request merged**